src/hooks/twilio.ts

```diff
@@ -133,8 +144,118 @@ export const useParticipantState = ({
     setMuted,
     toggleMuted,
 
-    isVideoHidden,
-    setVideoHidden,
-    toggleVideoHidden,
+    isVideoShown,
+    hideVideo,
+    showVideo,
+    toggleVideo,
+  };
+};
+
+export interface UseVideoRoomStateProps {
+  userId?: string;
+  roomName?: string;
+  showVideoByDefault?: boolean;
+}
+
+export const useVideoRoomState = ({
+  userId,
+  roomName,
+  showVideoByDefault = true,
+}: UseVideoRoomStateProps) => {
+  const [token, setToken] = useState<string>();
+
+  useEffect(() => {
+    (async () => {
+      if (!userId || !!token || !roomName) return;
+
+      const response = await getVideoToken({
+        userId,
+        roomName,
+      });
+
+      if (!response) return;
+
+      setToken(response.data.token);
+    })();
+  }, [userId, roomName, token]);
```

Comment:
> This would be less verbose if we just used Promise chain API rather than async/await.
> 
> We should also be exposing error information to the UI in some way from this

Done in https://github.com/sparkletown/sparkle/commit/720725dfbb2a8f1d9ad4ebb6d3b443b170f46024

> Might be worth looking at react-use's hooks here (eg. useAsync/useAsyncRetry):


This PR is already too big enough to add any new concepts. Marking this as resolved

Reply:
[NB]

> This PR is already too big enough to add any new concepts. 

We already have `react-use` in our dependencies. This would have resulted in less overall code required..?

Reply:
[probably blocking]

> We should also be exposing error information to the UI in some way from this.

This part doesn't look like it was done though? Which would be simplified by using the `useAsync`/etc things mentioned above (which is why I mentioned them)